src/driver.cpp

```diff
@@ -12,6 +12,31 @@ extern bpftrace::location loc;
 
 namespace bpftrace {
 
+namespace {
+std::string_view parse_license()
+{
+  std::string_view program = Log::get().get_source();
+  constexpr std::string_view spdx_id_marker = "SPDX-License-Identifier: ";
+  auto spdx_id_pos = program.find(spdx_id_marker);
+  if (spdx_id_pos == program.npos)
+    return {};
+
+  auto license_pos = spdx_id_pos + spdx_id_marker.size();
+  auto eol_pos = program.find("\n", spdx_id_pos + 1);
+  if (eol_pos == program.npos)
+    return {};
+
+  auto license = program.substr(license_pos, eol_pos - license_pos);
+
+  // Try to translate some known GPL-v2-compatible licenses into a format
+  // understood by the Linux kernel
+  if (license.find("GPL-2.0") == 0)
+    return "GPL";
```

Comment:
Nit: but maybe this should return "GPL v2"

Reply:
https://docs.kernel.org/process/license-rules.html#id1

The kernel docs description of "GPL v2" is:
> Same as “GPL”. It exists for historic reasons.

I took that to mean that "GPL" is the preferred format.